### PR TITLE
ARROW-4269: [Python] Fix serialization in pandas 0.22

### PR DESCRIPTION
**Pull request by pcmoritz**

### CODE REVIEW

File: python/pyarrow/serialization.py [hunk 1]
```diff
@@ -174,21 +174,21 @@ def _deserialize_pandas_series(data):
         custom_serializer=_pickle_to_buffer,
         custom_deserializer=_load_pickle_from_buffer)
 
-    if hasattr(pd.core.arrays, 'interval'):
+    if hasattr(pd.core, 'arrays') and hasattr(pd.core.arrays, 'interval'):
```
[the remark this answers is not in the document]
Comment (pcmoritz) [reply python/pyarrow/serialization.py]: Thanks, I fixed it!

**Comment by kszucs** [comment]
We're going to need this in the release. Thanks @pcmoritz!

**Comment by pcmoritz** [comment]
Does anybody know what happened to the code coverage? The changes can't possibly be related to the PR.

**Comment by pcmoritz** [comment]
the same seems to have happend to https://github.com/apache/arrow/pull/3387, so it's probably a fluke

**Comment by robertnishihara** [comment]
This looks good to me. I assume the change in coverage is due to the fact that it's a pure Python PR and so most tests didn't run.

**Pull request closed without merging**